### PR TITLE
Storage Write API Adds SinkTask Methods (start, put, stop)

### PR DESCRIPTION
**Pull request by b-goyal**
This PR adds
-> Task start for creating default stream 
-> Task put to send records via new api if configured
-> Task stop for cleaning up resources
-> Corresponding unit tests
-> Fixes unit tests failure due to newly added validations

JIRA: CC-19795

Epic: CC-19644

### CODE REVIEW

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 1]
```diff
@@ -206,14 +220,28 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
...
         ));
       }
       if (sanitize) {
         tableName = FieldNameSanitizer.sanitizeName(tableName);
       }
     }
 
+    return new String[]{dataset, tableName};
+  }
+  private PartitionedTableId getStorageApiRecordTable(String topic) {
```
Comment (aniketshrimal) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java:233]: nit: add a newline
```suggestion

  private PartitionedTableId getStorageApiRecordTable(String topic) {
```

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 2]
```diff
@@ -206,14 +220,28 @@ private PartitionedTableId getRecordTable(SinkRecord record) {
...
+    return topicToPartitionTableId.computeIfAbsent(topic, topicName -> {
+      String project = config.getString(BigQuerySinkConfig.PROJECT_CONFIG);
+      String[] datasetAndtableName = getDataSetAndTableName(topicName);
+      return new PartitionedTableId.Builder(TableId.of(project, datasetAndtableName[0], datasetAndtableName[1])).build();
+    });
+
+  }
```
Comment (aniketshrimal) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java:240]: nit: add a newline
```suggestion
  }
  
```

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 3]
```diff
@@ -265,10 +293,17 @@ public void writeSinkRecords(Collection<SinkRecord> records) {
 
     for (SinkRecord record : records) {
       if (record.value() != null || config.getBoolean(BigQuerySinkConfig.DELETE_ENABLED_CONFIG)) {
-        PartitionedTableId table = getRecordTable(record);
+        PartitionedTableId table = useStorageApi ? getStorageApiRecordTable(record.topic()) : getRecordTable(record);
```
Comment (aniketshrimal) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java:296]: Just a suggestion - I think we should maintain consistency incase of methods fetching the same values. Either we should sent the `record` or `record.topic()` in both the methods.

**Review by aniketshrimal — COMMENTED** [review]
Thanks @b-goyal for the PR. Have some minor comments and suggestions.

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 3 above)
Comment (b-goyal) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java:296]: Though they are returning the same type of response, these are 2 different methods. And the built response has same type but content can be different. `getStorageApiRecordTable` does not require the entire record. If we send the SinkRecord we will get the topic only there.

Comment (aniketshrimal) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java:296]: Sure, this makes sense.

**Comment by b-goyal** [comment]
Thanks for the inputs @aniketshrimal , have updated the PR.

**Review by aniketshrimal — APPROVED** [review]
LGTM!

**Pull request merged**